src/guide/advanced/wasm.md

```diff
@@ -0,0 +1,124 @@
+
+You'd still need to use ISI from inside your WASM binary to do anything on-chain, but you can do other things, and write e.g. Rust instead of composing Generic `structs` inside a macro to get conditional logic. Moreover, as long as you fit within the limits of WASM runtime, and the provided libraries, you can do anything (and everything) you want.
+
+The drawback is that this process is a tad more involved than just writing the ISI using the client libraries.
+
+
+## Simple Rust Smart Contract
+
+WASM projects, just like any other binary in Rust need to be separate crates. Don't worry, it doesn't have to be big.
+
+To get started you might want to use
+```bash
+cargo new --lib
+```
+to create a new project. Yes, we need the `lib` type, more on that later.
+
+The `Cargo.toml` of the project should look something like this:
+
+```toml
+[package]
+name  = "smartcontract"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ['cdylib']
+
+[dependencies]
+iroha_wasm = { git = "https://github.com/hyperledger/iroha/", branch = "iroha2" }
+```
+
+Note that the crate type is `cdylib`. Most Rust code is linked in a non-portable architecture and OS-specific static manner, but WASM is a portable format. Since C ABI is the *lingua franca* of the programming world and there is no other stable Rust ABI (yet), Iroha relies on the C-linkage to generate WASM bindings. Thankfully, you don't need to worry about foreign function interface (FFI)-related problems like `unsafe`, `repr(C)`, padding, alignment etc., `iroha_wasm` takes care of all that for you.
+
+`iroha_wasm` is the crate that contains all of the bindings, macros and trait implementations that you'd need to write the program, most notably the `iroha_wasm` attribute macro. The crate also exposes our `data_model` which contains all of the basic ISI and types. `parity-scale-codec` is (as of today) the chosen serialisation format with a strong possibility of getting replaced with a different (custom) serialisation format in the near future, as it seems to dominate the binary size[^2].
+
+Now that we have the preliminaries nailed down, we get to write some code for our smart contract. In the `src/lib.rs` you should write the following:
+
+
+```rust
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+
+use iroha_wasm::data_model::prelude::*;
+
+#[iroha_wasm::iroha_wasm]
+fn smartcontract_entry_point(_account_id: AccountId) {
+    let query = QueryBox::FindAllDomains(FindAllDomains {});
+    let domains: Vec<Domain> = query.execute().try_into().unwrap();
+
+    for domain in domains {
+        let new_account_id = AccountId {
+            name: Name::new("mad_hatter").unwrap(),
+            domain_id: domain.id,
+        };
+
+        Instruction::Register(RegisterBox::new(NewAccount::new(new_account_id))).execute();
+    }
+}
+```
+
+`cargo run --release` will submit the instruction and run it for you (be sure to have a peer up).
```

Comment:
I guess a general alternative instruction on how just to obtain a complide wasm file may be useful as well (of I just got confused)

Reply:
Especially if we are about to expand SDK sections with the related samples